scripts/utils/pausingHelper.ts

```diff
@@ -0,0 +1,143 @@
+export interface PausingActionResult {
+  operation: string;
+  success: boolean;
+  txHash?: string;
+  error?: string;
+  actionRedundant?: boolean;
+}
+
+interface ErrorPredicate {
+  errorName: string;
+  errorSelector: string;
+}
+
+/**
+ * Performs an unpause operation with centralized error handling
+ */
+export async function performUnpause(
+  operationName: string,
+  isPaused: () => Promise<boolean>,
+  unpauseFunction: () => Promise<any>,
+  predicate: ErrorPredicate
+): Promise<PausingActionResult> {
+  return await performPausingAction(
+    "unpause",
+    operationName,
+    isPaused,
+    unpauseFunction,
+    predicate
+  );
+}
+
+/**
+ * Performs a pause operation with centralized error handling
+ */
+export async function performPause(
+  operationName: string,
+  isPaused: () => Promise<boolean>,
+  pauseFunction: () => Promise<any>,
+  predicate: ErrorPredicate
+): Promise<PausingActionResult> {
+  return await performPausingAction(
+    "pause",
+    operationName,
+    isPaused,
+    pauseFunction,
+    predicate
+  );
+}
+
+async function performPausingAction(
+  pausingAction: "pause" | "unpause",
+  operationName: string,
+  isPaused: () => Promise<boolean>,
+  pausingFunction: () => Promise<any>,
+  predicate: ErrorPredicate
+): Promise<PausingActionResult> {
+  const pauseName = pausingAction === "pause" ? "Pause" : "Unpause";
+  const isCurrentlyPaused = await isPaused()
+
+  if (isCurrentlyPaused) {
+    console.log(`${operationName} is currently paused.`);
+    if (pausingAction === "pause") {
+      console.log(`No action needed; ${operationName} is already paused.`);
+      return {
+        operation: `${operationName} Pause`,
+        success: true,
+        actionRedundant: true,
+      };
+    } else {
+      console.log(`${operationName} is paused, proceeding to unpause.`);
+    }
+  } else {
+    console.log(`${operationName} is currently not paused.`);
+    if (pausingAction === "unpause") {
+      console.log(`No action needed; ${operationName} is already unpaused.`);
+      return {
+        operation: `${operationName} Unpause`,
+        success: true,
+        actionRedundant: true,
+      };
+    } else {
+      console.log(`${operationName} is not paused, proceeding to pause.`);
+    }
+  }
```

Comment:
Can be reduced to something like:

```suggestion
  const shouldPause = pausingAction === "pause"
  const isCurrentlyPaused = await isPaused();
  
  const pauseName = shouldPause ? "Pause" : "Unpause";
  const isInDesiredState = shouldPause ? isCurrentlyPaused : !isCurrentlyPaused
  
  if (isInDesiredState) {
    console.log(`${operationName} is already ${pausedAction}d.`);
    return {
      operation: `${operationName} ${pauseName}`,
      success: true,
      actionRedundant: true,
    };
  }
```

Reply:
done in https://github.com/bane-labs/bridge-evm-contracts/pull/325/commits/1ba03529400cc8d83da8b2927ce308333e16e98c